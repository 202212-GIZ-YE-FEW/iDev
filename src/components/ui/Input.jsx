import React from "react";

const Input = (props) => {
    const {
        handleChange,
        value,
        id,
        name,
        type,
        isRequired = false,
        placeholder,
        label,
        inputFontSize = "3xl",
        inputFontWeight = "light",
        inputWidthSize = "w-full",
        radius = "md",
        ...rest
    } = props;

    const inputClasses = `${inputWidthSize} border border-solid self-center border-gray-300 px-[20px] py-[10px] min-w-0 text-${inputFontSize} font-${inputFontWeight} rounded-${radius}`;

    return (
        <>
            {label && (
                <label
                    className='font-normal   whitespace-nowrap flex:me-10 flex:self-center'
                    htmlFor={rest.id || rest.name}
                >
                    {label}
                </label>
            )}
            <input
                onChange={handleChange}
                value={value}
                id={id}
                name={name}
                type={type}
                required={isRequired}
                placeholder={placeholder}
                className={inputClasses}
            />
        </>
    );
};
export default Input;

// Example of using
{
    /* <Input
label="email"
 type="name"
 name="name"
 placeholder="name"
 inputWidthSize="w-full" 


/> */
}
