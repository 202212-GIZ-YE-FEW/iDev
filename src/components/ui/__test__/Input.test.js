import renderer from "react-test-renderer";

import Input from "../Input";

it("renders correctly", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
});
