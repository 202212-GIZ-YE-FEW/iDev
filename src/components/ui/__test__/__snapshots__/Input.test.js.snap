// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="w-full border border-solid self-center border-gray-300 px-[20px] py-[10px] min-w-0 text-3xl font-light rounded-md"
  required={false}
/>
`;
